Vagrantfile

```diff
@@ -18,6 +18,34 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   config.vm.synced_folder ".", "/vagrant", disabled: true
   config.vm.synced_folder ".", "/srv/zulip"
 
+  proxy_config_file = ENV['HOME']+"/.zulip-vagrant-config"
+  http_proxy = https_proxy = no_proxy = ""
+
+  if File.file?(proxy_config_file)
+    IO.foreach(proxy_config_file) do |line|
+      line.chomp!
+      key, value = line.split(nil, 2)
+      case key
+      when /^([#;]|$)/; # ignore line
+      when "HTTP_PROXY"; http_proxy = value
+      when "HTTPS_PROXY"; https_proxy = value
+      when "NO_PROXY"; no_proxy = value
+      end
```

Comment:
Since we are passing `nil` as the first argument split uses `whitespace` (implicit) as the delimiter. 

The config file expects the value to be separated by whitespace. 

```
HTTP_PROXY http://proxy_host:port
HTTPS_PROXY http://proxy_host:port
NO_PROXY localhost,127.0.0.1,.example.com
```
Ruby has built in support to YAML. How about using YAML?

http://stackoverflow.com/questions/21422494/reading-and-updating-yaml-file-by-ruby-code